### ISSUE TITLE
Could you provide the "ObjectMapper.treeToValue(TreeNode, TypeReference)" method?

### ISSUE THREAD

**Issue by fantasy0v0** [FasterXML/jackson-databind#4046]
### Is your feature request related to a problem? Please describe.

The situation now is that without this overloaded method, I need to write extra code to achieve the result I want.

```java
ObjectMapper mapper = new ObjectMapper();
ArrayNode node = mapper.createArrayNode();
node.add("123").add("456");
JavaType javaType = mapper.constructType(new TypeReference<List<String>>() {});
try {
    mapper.treeToValue(node, javaType);
} catch(Exception e) {
    e.printStackTrace();
}
```

### Describe the solution you'd like

Could you provide this method to simplify my code?

Like the code below.

```java
ObjectMapper mapper = new ObjectMapper();
ArrayNode node = mapper.createArrayNode();
node.add("123").add("456");
try {
    mapper.treeToValue(node, new TypeReference<List<String>>() {});
} catch(Exception e) {
    e.printStackTrace();
}
```

**Comment by cowtowncoder** [comment]
This seems like something that could be an easy PR for someone looking for things to tackle; will label as such.

**Comment by timnick-snow** [comment]
`convertValue` may helpful
```java
        ObjectMapper mapper = new ObjectMapper();
        ArrayNode node = mapper.createArrayNode();
        node.add("123").add("456");
        try {
            List<String> list = mapper.convertValue(node, new TypeReference<List<String>>() {
            });
        } catch(Exception e) {
            e.printStackTrace();
        }
```

**Comment by timnick-snow** [comment]
> > convertValue may helpful转换值可能有帮助
> 
> Are you suggesting method name changed to `convertValue`? Or is there more to it. Asking just to make things clear.您是否建议将方法名称更改为 `convertValue` ？或者还有更多的事情要做。询问只是为了把事情说清楚。
> 
> @timnick-snow

No. 'convertValue' is an existing method. When fromValue is 'JsonNode', it should be equivalent to 'treeToValue'. But I'm not sure if there is a performance difference between them at the moment, as 'convertValue' seems to need to be serialized first. When fromValue is 'JsonNode', can this step be omitted?

**Comment by cowtowncoder** [comment]
> > first. When fromValue is 'JsonNode', can this step be omitted?
> 
> I think not and there is a reason for it. I don't remember in which issue or PR, but has been a discussion around it.

Actually serialization part can in fact be avoided because it is possible to create `JsonParser` that directly traverses `JsonNode` hierarchy.
So there is this performance benefit; although even serialization for `convertValue()` can avoid actual JSON generation: it will only add values into `TokenBuffer`, from which `JsonParser` (token stream) can be accessed.
I don't think performance difference has been measured but I would expect `treeToValue()` to be slightly faster.

**Comment by cowtowncoder** [comment]
Fixed via #4056, closing. Will be in 2.16.0
